extensions/obviousAlexC/penPlus.js

```diff
@@ -0,0 +1,2621 @@
+
+          x4 += sprXoff;
+          y4 += sprYoff;
+
+          x1 *= dWidth;
+          y1 *= dHeight;
+
+          x2 *= dWidth;
+          y2 *= dHeight;
+
+          x3 *= dWidth;
+          y3 *= dHeight;
+
+          x4 *= dWidth;
+          y4 *= dHeight;
+
+          const Attribute_ID = "squareStamp_" + curTarget.id;
+
+          triangleAttributesOfAllSprites[Attribute_ID][2] = myAttributes[7];
+          triangleAttributesOfAllSprites[Attribute_ID][3] = myAttributes[8];
+          triangleAttributesOfAllSprites[Attribute_ID][4] = myAttributes[9];
+          triangleAttributesOfAllSprites[Attribute_ID][5] = myAttributes[11];
+          triangleAttributesOfAllSprites[Attribute_ID][7] = myAttributes[10];
+          triangleAttributesOfAllSprites[Attribute_ID][10] = myAttributes[7];
+          triangleAttributesOfAllSprites[Attribute_ID][11] = myAttributes[8];
+          triangleAttributesOfAllSprites[Attribute_ID][12] = myAttributes[9];
+          triangleAttributesOfAllSprites[Attribute_ID][13] = myAttributes[11];
+          triangleAttributesOfAllSprites[Attribute_ID][15] = myAttributes[10];
+          triangleAttributesOfAllSprites[Attribute_ID][18] = myAttributes[7];
+          triangleAttributesOfAllSprites[Attribute_ID][19] = myAttributes[8];
+          triangleAttributesOfAllSprites[Attribute_ID][20] = myAttributes[9];
+          triangleAttributesOfAllSprites[Attribute_ID][21] = myAttributes[11];
+          triangleAttributesOfAllSprites[Attribute_ID][23] = myAttributes[10];
+
+          triFunctions.drawTri(
+            gl.getParameter(gl.CURRENT_PROGRAM),
+            x1,
+            y1,
+            x2,
+            y2,
+            x3,
+            y3,
+            attrib.color4f,
+            Attribute_ID
+          );
+
+          triFunctions.drawTri(
+            gl.getParameter(gl.CURRENT_PROGRAM),
+            x1,
+            y1,
+            x3,
+            y3,
+            x4,
+            y4,
+            attrib.color4f,
+            Attribute_ID
+          );
+        }
+      )
+      .setFilter();
+
+    extension
+      .addBlock(
+        "stamp pen square with the texture of [tex]",
+        "squareTexDown",
+        Scratch.BlockType.COMMAND,
+        ({ tex }, util) => {
+          //Just a simple thing to allow for pen drawing
+          const curTarget = util.target;
+
+          let currentTexture = null;
+          if (penPlusCostumeLibrary[tex]) {
+            currentTexture = penPlusCostumeLibrary[tex].texture;
+          } else {
+            const costIndex = curTarget.getCostumeIndexByName(
+              Scratch.Cast.toString(tex)
+            );
+            if (costIndex >= 0) {
+              const curCostume = curTarget.sprite.costumes_[costIndex];
+              if (costIndex != curTarget.currentCostume) {
+                curTarget.setCostume(costIndex);
+              }
+
+              currentTexture =
+                renderer._allSkins[curCostume.skinId].getTexture();
```

Comment:
VSCode thinks you forgot the scale argument here.

Reply:
The actual vm does not seem to care about a scale argument. (Could an optional argument)